core/src/main/scala/org/apache/spark/SSLOptions.scala

```diff
@@ -102,6 +106,36 @@ private[spark] case class SSLOptions(
     }
   }
 
+  /*
+   * The supportedAlgorithms set is a subset of the enabledAlgorithms that
+   * are supported by the current Java security provider for this protocol.
+   */
+  private val supportedAlgorithms: Set[String] = {
+    var context: Option[SSLContext] = Some(SSLContext.getDefault)
```

Comment:
I find this confusing but maybe I'm missing something. You're allocating a default here to fall back to, only if there is an exception in the try block? Is that clearer to set in the exception handling path then? and can this ever be `None`, so, why `Option`?